src/helpers/courseSlots.js

```diff
@@ -28,15 +28,23 @@ exports.createCourseSlot = async (payload, user) => {
   return (new CourseSlot(payload)).save();
 };
 
-exports.updateCourseSlot = async (slotFromDb, payload) => {
+exports.updateCourseSlot = async (slotFromDb, payload, user) => {
   const hasConflicts = await exports.hasConflicts({ ...slotFromDb, ...payload });
   if (hasConflicts) throw Boom.conflict(translate[language].courseSlotConflict);
 
   const updatePayload = { $set: pickBy(payload) };
 
   if (!payload.step) updatePayload.$unset = { step: '' };
 
-  await CourseSlot.updateOne({ _id: slotFromDb._id }, updatePayload);
+  const dates = {
+    courseId: slotFromDb.courseId,
+    from: slotFromDb.startDate,
+    to: payload.startDate,
+  };
+  await Promise.all([
+    CourseHistoriesHelper.createHistoryOnSlotEdition(dates, pick(user, '_id')),
```

Comment:
du coup ici, il faut que tu slotFromDb, payload, et pick(user, '_id') et tu travailles ta logique dans `createHistoryOnSlotEdition`

Reply:
oui!

Reply:
dans la meme logique faut-il envoyer tout lobjet user et selectionner user._id dans le helper?

Reply:
non pas forcement 
en fait pour mon premier point, le fait qu'il y ait from et to c'est vraiment propre au model CourseHistory donc ca doit etre dans son service associé - courseSlots lui il s'en fout

Reply:
daccord c'est bon normalement!